### DISCUSSION TITLE
SEQ SDK is not passing a filter with a string that contains an apostrophe

### DISCUSSION THREAD

**Discussion by gagwithgaffer** [datalust/seq-tickets#1136] [Help] [answered]
Hello, I'm using your SDK for fetching logs from SEQ using some filters.

One of the values I'm searching for against a structured logging property contains an apostrophe in the string and I've traced it back to your SDK where it fails. How can I solve this?

Please see screenshot:

![image](https://user-images.githubusercontent.com/51385414/108283588-70a96d80-717b-11eb-82e4-31527c7984b9.png)

**Comment by nblumhardt** [comment, marked as answer]
Hi! To escape an apostrophe embedded in a string, you'll need to double it:

```
logSiteName = 'Oli''s house'
```

Some handy query syntax hints in https://github.com/datalust/seq-cheat-sheets, along with the main docs in https://docs.datalust.co/docs/the-seq-query-language.

Hope this helps!

**Reply by gagwithgaffer** [reply]
OK thanks

**Reply by gagwithgaffer** [reply]
Hello, I've been through the whole list of properties I'm trying to filter and I have similar issues with a further 4qty examples below:

"Source Context='WebApp_PLUGIN.Pages.Settings.IndexModel'" // could be either underscoe or decimal point, or both...
"ThreadId='1'" // not sure why numbers aren't working
"ProcessId='48'" // not sure why numbers aren't working
"ProcessName=' WebApp-PLUGIN'" // must be the hyphen char

Could you please point me to the documentation that covers the workaround for these as well please? 

I used the C# string.replace for the issue with the apostrophe and would likely use a similar approach for the others above but doubling up on characters doesn't seem to work for the others.

filterString = filterString.Replace("'", @"''");

Many thanks

**Reply by gagwithgaffer** [reply]
Ok I seem to have resolved it.

My first item, "Source Context='WebApp_PLUGIN.Pages.Settings.IndexModel'" notice I had a typo, should have been SourceContext without a space.

For the properties with the numbers, it looks like Serilog had written them as integers and not in text form, so I will have to remove the apostrophes that wrap around the numbers i.e. ThreadId=1

Thanks anyway
